### DISCUSSION TITLE
How to restore funds when abandonchannel used but tx got confirmed after a few days.

### DISCUSSION THREAD

**Discussion by Krtekcze** [lightningnetwork/lnd#8110] [Q&A] [answered]
Greetings guys,

 I find myself in a situation where a Bitcoin transaction has been confirmed, but the "abandonchannel" command has been executed. 

```
lnshell@x:~$ lncli pendingchannels | grep limbo
    "total_limbo_balance": "0",
lnshell@x:~$ lncli pendingchannels | grep channel_point
                "channel_point": "21da51b97b33abdaf172ea3c452676b58af96727ecdb926fb4b048bec24fc5c0:1",
lnshell@x:~$ lncli abandonchannel --funding_txid 21da51b97b33abdaf172ea3c452676b58af96727ecdb926fb4b048bec24fc5c0 --output_index 1 --i_know_what_i_am_doing
```

I have in my possession the output from the "lncli pendingchannels" command before I executed the abandonchannel command, the other peer confirmed he gave up on the tx approximately two weeks after its initial broadcast. The other peer is technical and cooperative.

```
lnshell@x:~$ lncli pendingchannels
{
    "total_limbo_balance": "0",
    "pending_open_channels": [
        {
            "channel": {
                "remote_node_pub": "03c157946cc1cd376b929e36006e645fae490b1b1d4156b40db804e01b4bda48cd",
                "channel_point": "21da51b97b33abdaf172ea3c452676b58af96727ecdb926fb4b048bec24fc5c0:1",
                "capacity": "5000000",
                "local_balance": "4996530",
                "remote_balance": "0",
                "local_chan_reserve_sat": "50000",
                "remote_chan_reserve_sat": "50000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": ""
            },
            "confirmation_height": 0,
            "commit_fee": "2810",
            "commit_weight": "772",
            "fee_per_kw": "2500"
        }
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
    ]
}
lnshell@x:~$
```

I have taken precautions by maintaining several Secure Channel Protocol (SCP) backups; however, my concern lies in their capability to encompass unconfirmed open transactions. I kindly ask for your collective wisdom and insights regarding the potential avenues available for the restoration of these funds, or if, regrettably, they are irreversibly lost.

I believe Zombie Channel Recovery (https://github.com/lightninglabs/chantools/blob/master/doc/zombierecovery.md#zombie-channel-recovery) can be used to restore the funds, could you guys please confirm if this is the case?

Your assistance in this matter would be of great value. Thank you in advance for your time and contributions.

Best regards,
Krtekcze

**Comment by guggero** [comment, marked as answer]
Yes, you can use the zombie recovery.
You'll need the "match" file to get started, from the information you provided this would look like this:
```
{
  "node1": {
    "identity_pubkey": "03c157946cc1cd376b929e36006e645fae490b1b1d4156b40db804e01b4bda48cd",
    "contact": "your peer's node"
  },
  "node2": {
    "identity_pubkey": "03yyyyyy",
    "contact": "your node"
  },
  "channels": [
    {
      "short_channel_id": "0000",
      "chan_point": "21da51b97b33abdaf172ea3c452676b58af96727ecdb926fb4b048bec24fc5c0:1",
      "address": "bc1qp4j7mx4k08z6kgz4jed3qrdcy5dphfr809a7jx95g9cdanrc9qwqy4avm4",
      "capacity": 5000000
    }
  ]
}
```
You need to add your node's public key in the "node2 -> identity_pubkey" field, then save this as a `.json` file which you both can then use in the next steps as described in the guide above.

**Reply by Krtekcze** [reply]
I'm not sure if the other side is supposed to create match.json file with capacity 0 as the channel has never been operational so it would look like below:

```
{
  "node1": {
    "identity_pubkey": "03c157946cc1cd376b929e36006e645fae490b1b1d4156b40db804e01b4bda48cd",
    "contact": "The Continental"
  },
  "node2": {
    "identity_pubkey": "02f9695db875da0f1e732eeb1b3e8361ccecee2ecb2807f645bdbdf52086f05998",
    "contact": "KrtekLND"
  },
  "channels": [
    {
      "short_channel_id": "0000",
      "chan_point": "21da51b97b33abdaf172ea3c452676b58af96727ecdb926fb4b048bec24fc5c0:1",
      "address": "bc1qp4j7mx4k08z6kgz4jed3qrdcy5dphfr809a7jx95g9cdanrc9qwqy4avm4",
      "capacity": 0
    }
  ]
}
```

Could you please confirm? 
Thank you

**Reply by guggero** [reply]
The other side needs to use the same match.json file.

**Comment by Krtekcze** [comment]
The transaction has been successfully signed by both sides and broadcasted to the mempoool. Thank you @guggero  very much for your help.

https://mempool.space/cs/tx/f25fe65a2106e9ad88cfd7fe228d681cc6af4ad62d74631624877f3277ac12ef
